hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/TargetAssignmentOperations.java

```diff
@@ -0,0 +1,285 @@
+                    .assignDistributionSet(mapEntry.getKey(),
+                            tempIdList.stream().map(t -> maintenanceWindowLayout.isEnabled()
+                                    ? new TargetWithActionType(t.getControllerId(), actionType, forcedTimeStamp,
+                                            maintenanceSchedule, maintenanceDuration, maintenanceTimeZone)
+                                    : new TargetWithActionType(t.getControllerId(), actionType, forcedTimeStamp))
+                                    .collect(Collectors.toList()));
+            if (distributionSetAssignmentResult.getAssigned() > 0) {
+                notification.displaySuccess(
+                        i18n.getMessage("message.target.assignment", distributionSetAssignmentResult.getAssigned()));
+            }
+            if (distributionSetAssignmentResult.getAlreadyAssigned() > 0) {
+                notification.displaySuccess(i18n.getMessage("message.target.alreadyAssigned",
+                        distributionSetAssignmentResult.getAlreadyAssigned()));
+            }
+        }
+        refreshPinnedDetails(saveAssignedList, managementUIState, eventBus, eventSource);
+        managementUIState.getAssignedList().clear();
+        notification.displaySuccess(i18n.getMessage("message.target.ds.assign.success"));
+        eventBus.publish(eventSource, SaveActionWindowEvent.SAVED_ASSIGNMENTS);
+    }
+
+    private static void refreshPinnedDetails(final Map<Long, List<TargetIdName>> saveAssignedList,
+            final ManagementUIState managementUIState, final UIEventBus eventBus, final Object eventSource) {
+        final Optional<Long> pinnedDist = managementUIState.getTargetTableFilters().getPinnedDistId();
+        final Optional<TargetIdName> pinnedTarget = managementUIState.getDistributionTableFilters().getPinnedTarget();
+
+        if (pinnedDist.isPresent()) {
+            if (saveAssignedList.keySet().contains(pinnedDist.get())) {
+                eventBus.publish(eventSource, PinUnpinEvent.PIN_DISTRIBUTION);
+            }
+        } else if (pinnedTarget.isPresent()) {
+            final Set<TargetIdName> assignedTargetIds = managementUIState.getAssignedList().keySet();
+            if (assignedTargetIds.contains(pinnedTarget.get())) {
+                eventBus.publish(eventSource, PinUnpinEvent.PIN_TARGET);
+            }
+        }
+    }
+
+    /**
+     * Check wether the maintenance window is valid or not
+     * 
+     * @param maintenanceWindowLayout
+     *            the maintenance window layout
+     * @param log
+     *            the logger for the calling class
+     * @param notification
+     *            the UI Notification
+     * @return boolean if maintenance window is valid or not
+     */
+    public static boolean isMaintenanceWindowValid(final MaintenanceWindowLayout maintenanceWindowLayout,
+            final Logger log, final UINotification notification) {
```

Comment:
add static logger to class and remove Logger argument from method (LoggerFactory.getLogger(TargetAssignmentOperations.class))